codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttCodecUtil.java

```diff
@@ -43,22 +43,28 @@ static boolean isValidClientId(MqttVersion mqttVersion, String clientId) {
             return clientId != null && clientId.length() >= MIN_CLIENT_ID_LENGTH &&
                 clientId.length() <= MAX_CLIENT_ID_LENGTH;
         }
-        if (mqttVersion == MqttVersion.MQTT_3_1_1) {
+        if (mqttVersion == MqttVersion.MQTT_3_1_1 || mqttVersion == MqttVersion.MQTT_5) {
             // In 3.1.3.1 Client Identifier of MQTT 3.1.1 specification, The Server MAY allow ClientId’s
             // that contain more than 23 encoded bytes. And, The Server MAY allow zero-length ClientId.
             return clientId != null;
         }
         throw new IllegalArgumentException(mqttVersion + " is unknown mqtt version");
     }
 
-    static MqttFixedHeader validateFixedHeader(MqttFixedHeader mqttFixedHeader) {
+    static MqttFixedHeader validateFixedHeader(MqttFixedHeader mqttFixedHeader, MqttVersion mqttVersion) {
```

Comment:
Did you check for API breakage?

Reply:
This method has package visibility and not intended to be called by classes outside of MQTT coded. `MqttDecoder.decodeFixedHeader` is the only place where it's called. Therefore, changing the API shouldn't be an issue here.

Reply:
AS @paul-lysak noted... this is fine